rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyIdentified.java

```diff
@@ -110,41 +111,48 @@ public int getCount()
     @Override
     public void perform(GraphRewrite event, EvaluationContext context, WindupVertexFrame payload)
     {
-        ProjectModel project;
+        Set<ProjectModel> projects = new HashSet<>();
         if (payload instanceof ProjectModel)
         {
-            project = (ProjectModel) payload;
+            projects.add((ProjectModel) payload);
         }
         else if (payload instanceof FileModel)
         {
-            project = ((FileModel) payload).getProjectModel();
+            projects.add(((FileModel) payload).getProjectModel());
         } else if (payload instanceof FileLocationModel)
         {
-            project = ((FileLocationModel)payload).getFile().getProjectModel();
+            projects.add(((FileLocationModel)payload).getFile().getProjectModel());
+        } else if (payload instanceof JNDIResourceModel)
+        {
+            JNDIResourceModel jndiResourceModel = (JNDIResourceModel)payload;
+            jndiResourceModel.getApplications().forEach(projects::add);
         }
         else
         {
             return;
```

Comment:
@jsight Should we create `else if` branch for each model used for technology report? Because right now, writing rules for different models then those you have listed there won't have any effect. It will always end up in 
```java
else { return }
```
branch.

If so, is the only purpose of this code to get project model of the entity? There is method `getRootProjectModels` in `BelongsToProject` interface which is created exactly for this purpose

Reply:
it seems like that see @mrizzi PR i merged https://github.com/windup/windup/pull/1173

Reply:
@klinki are you suggesting to use 
```java
((BelongsToProject) payload).getRootProjectModels().forEach(projects::add);
```
instead of the `if-else` chain?

Reply:
Yes

Reply:
I made PR: https://github.com/windup/windup/pull/1174 with BelongsToProject

Reply:
Yeah, I think that makes sense too. I wasn't thinking about how many variations of project references there were when I added this.

Although, how does that work when the root project is "shared-libs"?